release/authorization/README.md

```diff
@@ -0,0 +1,420 @@
+
+```shell
+    /this/is/a/message_path
+```
+
+Additionally, "stevie" is permitted access to all paths below the defined path,
+in a READ only mode, such as:
+
+```shell
+    /this/is/a/message_path/the
+    /this/is/a/message_path/the/one
+    /this/is/a/message_path/the/one/that
+    /this/is/a/message_path/the/one/that/knocks
+```
+
+The second policy rule:
+
+```proto
+# Action memberss of family-group to run /this/is/a/different/message_path
+policy {
+  id: "two"
+  log_level: LOG_BRIEF
+  path {
+    origin: "foo"
+    elem { name: "this" }
+    elem { name: "is" }
+    elem { name: "a" }
+    elem { name: "different" }
+    elem { name: "message_path" }
+  }
+  action: PERMIT
+  level: READ
+  level: WRITE
+  group { name: "family-group" }
+}
+```
+
+example policy permits members or the family-group access to a single path, for
+reading or writing:
+
+```shell
+    /this/is/a/different/message_path
+```
+
+and all path elements beyond "message_path":
+
+```shell
+    /this/is/a/different/message_path/foo
+    /this/is/a/different/message_path/bar
+    /this/is/a/different/message_path/foo/baz/bing/boop
+```
+
+The third policy rule:
+
+```proto
+# Demonstrate a key with an attribute defined.
+policy {
+  id: "key"
+  log_level: LOG_BRIEF
+  path {
+    origin: "foo"
+    elem { name: "this" }
+    elem { name: "is" }
+    elem { name: "a" }
+    elem {
+      name: "keyed"
+      key{
+        key: "name"
+        value: "Ethernet1/2/3"
+      }
+    }
+    elem { name: "message_path" }
+  }
+  action: PERMIT
+  level: READ
+  group { name: "test-group" }
+}
+```
+
+Permits access by the "test-group" users to the keyed path, in a
+read only manner:
+
+```shell
+    /this/is/a/keyed[name=Ethernet1/2/3]/message_path
+```
+
+and all path elementas as beyond "message_path". The final policy rule:
+
+```proto
+# Demonstrate a key with a wildcard attribute.
+policy {
+  id: "wyld"
+  log_level: LOG_BRIEF
+  path {
+    origin: "foo"
+    elem { name: "this" }
+    elem { name: "is" }
+    elem { name: "a" }
+    elem {
+      name: "keyed"
+      key{
+        key: "name"
+        value: "*"
+      }
+    }
+    elem { name: "message_path" }
+  }
+  action: PERMIT
+  level: READ
+  group { name: "family-group" }
+}
+```
+
+permits access by the "family-group" users to the keyed path, with no
+restrictions on the key values, but still as read-only:
+
+```shell
+    /this/is/a/keyed[name=Ethernet1/2/3]/message_path
+    /this/is/a/keyed[name=POS3]/message_path
+    /this/is/a/keyed[name=Serial4/1]/message_path
+    /this/is/a/keyed[name=HSSI2]/message_path
+```
+
+Additionally, the path elements beyond "message_path" are available for access
+to this group as well.
+
+The wildcard character "*" (asterisk) is only able to be used as a value in
+keyed elements, if the keys are missing in a keyed path a wildcard is assumed.
+The wildcard is only used to mask out all possible values but not portions of
+values, for instance:
```

Comment:
do we need to say anything about paths with multiple attributes?  presumably it's fine to wildcard any combination of keys in such a path?

Reply:
I think this came up previously, I don't see why we'd limit the number of keyed parts in a rule.
The existing text doesn't talk about one or many  keyed parts, so I'd just leave it as is.